plugins/haconfig/keepalived/generator.go

```diff
@@ -0,0 +1,172 @@
+
+	ports := make([]kapi.ContainerPort, 1)
+	ports[0] = kapi.ContainerPort{
+		ContainerPort: options.ServicePort,
+		HostPort:      options.ServicePort,
+	}
+
+	mounts := make([]kapi.VolumeMount, 1)
+	mounts[0] = kapi.VolumeMount{
+		Name:      libModulesVolumeName,
+		ReadOnly:  true,
+		MountPath: libModulesPath,
+	}
+
+	return &kapi.Container{
+		Name:            containerName,
+		Image:           image,
+		Ports:           ports,
+		Privileged:      true,
+		ImagePullPolicy: kapi.PullIfNotPresent,
+		VolumeMounts:    mounts,
+		Env:             env.List(),
+	}
+}
+
+//  Generate the HA failover monitor (keepalived) container configuration.
+func generateContainerConfig(name string, options *haconfig.HAConfigCmdOptions) []kapi.Container {
+	containers := make([]kapi.Container, 0)
+
+	if len(options.VirtualIPs) < 1 {
+		return containers
+	}
+
+	config := getClientConfig(options.Credentials)
+	env := generateEnvEntries(name, options, config)
+
+	c := generateFailoverMonitorContainerConfig(name, options, env)
+	if c != nil {
+		containers = append(containers, *c)
+	}
+
+	return containers
+}
+
+//  Generate the HA failover monitor (keepalived) container volume config.
+func generateVolumeConfig() []kapi.Volume {
+	hostPath := &kapi.HostPathVolumeSource{Path: libModulesPath}
+	src := kapi.VolumeSource{HostPath: hostPath}
+
+	vol := kapi.Volume{Name: libModulesVolumeName, VolumeSource: src}
+	return []kapi.Volume{vol}
+}
+
+//  Generate the HA deployment template.
+func generateDeploymentTemplate(name string, options *haconfig.HAConfigCmdOptions, selector map[string]string) dapi.DeploymentTemplate {
+	podTemplate := &kapi.PodTemplateSpec{
+		ObjectMeta: kapi.ObjectMeta{Labels: selector},
+		Spec: kapi.PodSpec{
+			HostNetwork: true,
+			Containers:  generateContainerConfig(name, options),
+			Volumes:     generateVolumeConfig(),
+		},
+	}
+
+	return dapi.DeploymentTemplate{
+		Strategy: dapi.DeploymentStrategy{
+			Type: dapi.DeploymentStrategyTypeRecreate,
+		},
+		ControllerTemplate: kapi.ReplicationControllerSpec{
+			// TODO: v0.1 requires a manual resize to the
+			//       replicas to match current cluster state.
+			//       v0.1+ could do this with either a watcher
+			//       that updates the replica count or better
+			//       yet, some way to kubernetes to say run
+			//       pods on each and every node that matches
+			//       the selector.
+			Replicas: options.Replicas,
+			Selector: selector,
+			Template: podTemplate,
+		},
+	}
+}
+
+//  Generate the HA deployment configuration.
+func GenerateDeploymentConfig(name string, options *haconfig.HAConfigCmdOptions, selector map[string]string) *dapi.DeploymentConfig {
+
+	return &dapi.DeploymentConfig{
+		ObjectMeta: kapi.ObjectMeta{
+			Name:   name,
+			Labels: selector,
+		},
+		Triggers: []dapi.DeploymentTriggerPolicy{
+			{Type: dapi.DeploymentTriggerOnConfigChange},
+		},
+		Template: generateDeploymentTemplate(name, options, selector),
```

Comment:
Doesn't seem to make it any more readable to break it out - recommend just adding the template here.